src/spikeinterface/curation/train_manual_curation.py

```diff
@@ -0,0 +1,796 @@
+        if self.metric_names is None:
+            warnings.warn("No metric_names provided, using all metrics calculated by the analyzers")
+            self.metric_names = self.testing_metrics.columns.tolist()
+
+        conflicting_metrics = self._check_metrics_parameters(analyzers, enforce_metric_params)
+
+        self.metrics_params = {}
+        if analyzers[0].has_extension("quality_metrics") is True:
+            self.metrics_params["quality_metric_params"] = analyzers[0].extensions["quality_metrics"].params
+
+            # Only save metric params which are 1) consistent and 2) exist in metric_names
+            qm_names = self.metrics_params["quality_metric_params"]["metric_names"]
+            consistent_metrics = list(set(qm_names).difference(set(conflicting_metrics)))
+            consistent_metric_params = {
+                metric: analyzers[0].extensions["quality_metrics"].params["qm_params"][metric]
+                for metric in consistent_metrics
+            }
+            self.metrics_params["quality_metric_params"]["qm_params"] = consistent_metric_params
+
+        if analyzers[0].has_extension("template_metrics") is True:
+            self.metrics_params["template_metric_params"] = analyzers[0].extensions["template_metrics"].params
+            if "template_metrics" in conflicting_metrics:
+                self.metrics_params["template_metric_params"] = analyzers[0].extensions["template_metrics"].params = {}
+
+        self.process_test_data_for_classification()
+
+    def _check_metrics_parameters(self, analyzers, enforce_metric_params):
+        """Checks that the metrics of each analyzer have been calcualted using the same parameters"""
+
+        conflicting_metrics = []
+        for analyzer_index_1, analyzer_1 in enumerate(analyzers):
+            for analyzer_index_2, analyzer_2 in enumerate(analyzers):
+
+                if analyzer_index_1 <= analyzer_index_2:
+                    continue
+                else:
+
+                    qm_params_1 = {}
+                    qm_params_2 = {}
+                    tm_params_1 = {}
+                    tm_params_2 = {}
+
+                    if analyzer_1.has_extension("quality_metrics") is True:
+                        qm_params_1 = analyzer_1.extensions["quality_metrics"].params["qm_params"]
+                    if analyzer_2.has_extension("quality_metrics") is True:
+                        qm_params_2 = analyzer_2.extensions["quality_metrics"].params["qm_params"]
+                    if analyzer_1.has_extension("template_metrics") is True:
+                        tm_params_1 = analyzer_1.extensions["template_metrics"].params["metrics_kwargs"]
+                    if analyzer_2.has_extension("template_metrics") is True:
+                        tm_params_2 = analyzer_2.extensions["template_metrics"].params["metrics_kwargs"]
+
+                    conflicting_metrics_between_1_2 = []
+                    # check quality metrics params
```

Comment:
How similar is this function to `check_metric_names_are_the_same`?

Reply:
Not very similar, because this is checking the parameters used to compute the metrics.